StackExchange.Redis/StackExchange/Redis/RedisDatabase.cs

```diff
@@ -1579,6 +1579,605 @@ public IEnumerable<SortedSetEntry> SortedSetScan(RedisKey key, RedisValue patter
+
+            return ExecuteSync(msg, ResultProcessor.Int64);
+        }
+
+        public Task<long> StreamMessagesDeleteAsync(RedisKey key, string[] messageIds, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database,
+                flags,
+                RedisCommand.XDEL,
+                key,
+                messageIds.ToRedisValueArray());
+
+            return ExecuteAsync(msg, ResultProcessor.Int64);
+        }
+
+        public StreamPendingInfo StreamPendingInfoGet(RedisKey key, RedisValue groupName, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database, flags, RedisCommand.XPENDING, key, groupName);
+            return ExecuteSync(msg, ResultProcessor.StreamPendingInfo);
+        }
+
+        public Task<StreamPendingInfo> StreamPendingInfoGetAsync(RedisKey key, RedisValue groupName, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database, flags, RedisCommand.XPENDING, key, groupName);
+            return ExecuteAsync(msg, ResultProcessor.StreamPendingInfo);
+        }
+
+        public StreamPendingMessageInfo[] StreamPendingMessageInfoGet(RedisKey key, RedisValue groupName, RedisValue minId, RedisValue maxId, int count, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamPendingMessagesMessage(key, groupName, minId, maxId, count, RedisValue.Null, flags);
+            return ExecuteSync(msg, ResultProcessor.StreamPendingMessages);
+        }
+
+        public StreamPendingMessageInfo[] StreamPendingMessageInfoGet(RedisKey key, RedisValue groupName, RedisValue minId, RedisValue maxId, int count, RedisValue consumerName, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamPendingMessagesMessage(key, groupName, minId, maxId, count, consumerName, flags);
+            return ExecuteSync(msg, ResultProcessor.StreamPendingMessages);
+        }
+
+        public Task<StreamPendingMessageInfo[]> StreamPendingMessageInfoGetAsync(RedisKey key, RedisValue groupName, RedisValue minId, RedisValue maxId, int count, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamPendingMessagesMessage(key, groupName, minId, maxId, count, RedisValue.Null, flags);
+            return ExecuteAsync(msg, ResultProcessor.StreamPendingMessages);
+        }
+
+        public Task<StreamPendingMessageInfo[]> StreamPendingMessageInfoGetAsync(RedisKey key, RedisValue groupName, RedisValue minId, RedisValue maxId, int count, RedisValue consumerName, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamPendingMessagesMessage(key, groupName, minId, maxId, count, consumerName, flags);
+            return ExecuteAsync(msg, ResultProcessor.StreamPendingMessages);
+        }
+
+        public RedisStreamEntry[] StreamRange(RedisKey key, RedisValue minId, RedisValue maxId, CommandFlags flags = CommandFlags.None)
```

Comment:
Same for below - instead of 2 methods - how about we make `count` an `int?` on these? That'd take us from 4 (sync+async) methods down to 2.